maven go-offline failure because of "was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced"

example https://github.com/trinodb/trino/actions/runs/4794853359/jobs/8528698799?pr=17092 ([maven-checks (17)](https://github.com/trinodb/trino/actions/runs/4794853359/jobs/8528698799?pr=17092#logs))

```
2023-04-25T11:27:28.2818232Z + .github/bin/retry ./mvnw -B dependency:go-offline -Dsilent
2023-04-25T11:27:28.2866400Z + ./mvnw -B dependency:go-offline -Dsilent
...
2023-04-25T11:28:19.3603614Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.5.0:go-offline (default-cli) on project trino-record-decoder: org.eclipse.aether.resolution.DependencyResolutionException: The following artifacts could not be resolved: io.confluent:kafka-protobuf-provider:jar:7.3.1 (absent): Could not find artifact io.confluent:kafka-protobuf-provider:jar:7.3.1 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
...
2023-04-25T11:28:29.3809131Z + ./mvnw -B dependency:go-offline -Dsilent
...
2023-04-25T11:28:45.8867438Z [ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.5.0:go-offline (default-cli) on project trino-record-decoder: org.eclipse.aether.resolution.DependencyResolutionException: The following artifacts could not be resolved: io.confluent:kafka-protobuf-provider:jar:7.3.1 (absent): Failure to find io.confluent:kafka-protobuf-provider:jar:7.3.1 in https://repo.maven.apache.org/maven2 was cached in the local repository, resolution will not be reattempted until the update interval of central has elapsed or updates are forced -> [Help 1]
```

cc @nineinchnick

--- a/lib/trino-record-decoder/pom.xml
+++ b/lib/trino-record-decoder/pom.xml
@@ -16,6 +16,16 @@
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
     </properties>
 
+    <repositories>
+        <repository>
+            <id>confluent</id>
+            <url>https://packages.confluent.io/maven/</url>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+        </repository>
+    </repositories>
+
     <dependencies>
         <!-- Trino SPI -->
         <dependency>

